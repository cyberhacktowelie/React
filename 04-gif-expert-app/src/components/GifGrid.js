// import React, { useState, useEffect } from 'react';
import { PropTypes } from 'prop-types';

import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    // const [ images, setImages ] = useState([]);

    const { data:images , loading } = useFetchGifs(category);

    // useEffect(() => {
    //     // getGifs();
    //     // getGifs(category).then(img => setImages(img));  
    //     getGifs(category).then(setImages);
    // }, [ category ]);


    // const apiKey = 'QKXFCiGkEFDCSb1fyIPKGNNZ2wIx1E5g';
    // const getGifs = async () => {
    //     const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI(category) }&limit=10&api_key=${apiKey}`;
    //     const resp = await fetch(url);
    //     const { data } = await resp.json();

    //     const gifs = data.map(img => {
    //         return {
    //             id: img.id,
    //             title: img.title,
    //             url: img.images?.downsized_medium.url
    //         }});

    //     // console.log(gifs);
    //     setImages(gifs);
    // };

    // getGifs();

    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {/* { loading ? <p>Cargando...</p> : null } */}
            { loading && <p className="animate__animated animate__flash">Cargando...</p> }
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridItem key={ img.id} {...img} />
                    ))
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}