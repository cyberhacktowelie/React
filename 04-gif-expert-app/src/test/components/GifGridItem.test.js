import { GifGridItem } from '../../components/GifGridItem';
import { shallow } from 'enzyme';

describe('Pruebas en el componente GifGridItem   GifGridItem.test.js', () => {

    const title = 'Titulo';
    const url = 'https://media.giphy.com/media/3o7btLwvZuXyj3N3zS/giphy.gif';
    let wrapper = shallow(<GifGridItem 
        title={title}
        url={url} />);

    beforeEach(() => {
        wrapper = shallow(<GifGridItem 
            title={title}
            url={url} />);
    });

    test('El itemm debe renderizarse correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    })

    test('Debe de tener un parrafo con el title', () => {
            
            const p = wrapper.find('p');
            expect( p.text() ).toBe(title);
    
    })

    test('Debe de tener una imagen con la url y alt', () => {
                
                const img = wrapper.find('img');
                // expect( img.prop('src') ).toBe(url);
                const { alt, src } = img.props();
                expect( alt ).toBe(title);
                expect( src ).toBe(url);
                
    })

    test('Debe de tener una clase animate__animated animate__fadeIn', () => {
                
                const div = wrapper.find('div');
                const { className } = div.props();
                expect( className ).toContain('animate__animated');
                expect( className.includes('animate__fadeIn') ).toBe(true);
                expect( div.hasClass('animate__animated') ).toBe(true);
                expect( div.hasClass('animate__fadeIn') ).toBe(true);
                
    })


});