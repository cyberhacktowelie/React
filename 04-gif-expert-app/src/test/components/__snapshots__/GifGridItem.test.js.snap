// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem   GifGridItem.test.js El itemm debe renderizarse correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://media.giphy.com/media/3o7btLwvZuXyj3N3zS/giphy.gif"
  />
  <p>
    Titulo
  </p>
</div>
`;
